### ISSUE TITLE
get_products_in_view() causes MySQL errors with large # of products

### ISSUE THREAD

**Issue by dbarproductions** [woothemes/woocommerce#7664]
This function, in the WC_Query class, removes all filters and attempts to store an array of ALL product ids in a transient option, for what purpose, it's not really clear.

We are testing with 70,000 products, and with Debug on we get:
WordPress database error: [MySQL server has gone away] and then it shows the content of this transient (all 70,000 items) that it was trying to save.

Why do we need this? Can we at least include some kind of filter/option to turn it off?  For now, we have to bypass the shop page entirely and make our own custom shop page because of this and because the normal query for products takes too long with this many products.

**Comment by mikejolley** [comment]
We'll look at this for 2.4. The reason this is done is because the layered nav + price widgets need to show counts/values for the 'view', not the entire catalog. The ids are cross referenced against IDs of objects in each term for example.

**Comment by dbarproductions** [comment]
Thanks Mike.

It was killing MySQL on my little notebook development machine because it was trying to index all 70,000 products.  It would probably work on the real server, but it was still too slow. Our solution for now was to create a separate page that we don't set up as a product archive so that there are no Wordpress & Woocommerce queries happening on the whole catalog, and that function doesn't get called.  That seems to be the only solution for now, and it's much faster anyway. We set up our own custom table for the search/filtering system we developed so that we don't have to do any multi-table joins or index all the products... works much faster with the number of products we are dealing with since we find all the results from one optimized table and then simply grab the needed product info for the small set of products actually shown on screen.
